lib/core/app/app_router.dart

```diff
@@ -87,9 +87,17 @@ class AppRouter {
           path: Pages.chooseAmountSlider.path,
           name: Pages.chooseAmountSlider.name,
           builder: (context, state) {
-            final flow =
-                state.extra != null ? state.extra as Flows : Flows.main;
-            return ChooseAmountSliderScreen(flow: flow);
+            Flows flow;
+            final bool isCoinFlow =
+                state.uri.queryParameters['isCoinFlow']?.contains('true') ??
+                    false;
+            if (isCoinFlow) {
+              flow = Flows.coin;
+              return ChooseAmountSliderScreen(flow: flow);
+            } else {
```

Comment:
I wanted to pass the flow as an extra to keep consistency, but adding extra data when redirecting is still an open issue on the [go router package.](https://github.com/flutter/flutter/issues/105294)